lib/ukdebug/gdbstub.c

```diff
@@ -998,6 +1038,34 @@ static int gdb_main_loop(struct gdb_excpt_ctx *g)
 	return (int)r;
 }
 
+static __ssz gdb_virt_out(struct uk_console *dev __unused, const char *str,
+			  __sz len)
+{
+	unsigned long irqs;
+
+	if (str) {
+		/* TODO: SMP Support */
+		irqs = ukplat_lcpu_save_irqf();
```

Comment:
Why disabling interrupts and would it make more sense to do this within `gdb_send_message_packet()` and company?

Reply:
As discussed offline, we keep this as a safety measure. See the comment in the code now.

Reply:
Thanks for the comment.